indirect.h

```diff
@@ -178,21 +188,31 @@ class indirect {
       allocator_traits::is_always_equal::value) {
     if (this == &other) return *this;
 
-    if constexpr (allocator_traits::propagate_on_container_move_assignment::
-                      value) {
-      if (alloc_ != other.alloc_) {
-        reset();  // using current allocator.
-        alloc_ = other.alloc_;
-      }
-    }
-    reset();  // We may not have reset above and it's a no-op if valueless.
+    // Check to see if the allocators need to be updated.
+    // We defer actually updating the allocator until later because it may be
+    // needed to delete the current control block.
+    bool update_alloc =
+        allocator_traits::propagate_on_container_copy_assignment::value &&
```

Comment:
I think this should be `propagate_on_container_move_assignment` and not `propagate_on_container_copy_assignment` right?  Looks like that is what is was previously...
```suggestion
        allocator_traits::propagate_on_container_move_assignment::value &&
```